File: Temperature simulator/Temperature2.py
import matplotlib.pyplot as plt
from random import uniform, randint
SENSITIVITY = 0.5
MAX_TEMP = 8
MIN_TEMP = 4
Number = 0
T = randint(MIN_TEMP, MAX_TEMP)


T_values = [T]  # List to store temperature values

while Number < 100:
    newT=randint(MIN_TEMP, MAX_TEMP)
    while T >= newT - SENSITIVITY:
      T-=SENSITIVITY 
    while T<= newT - SENSITIVITY:
      T+=SENSITIVITY

    T_values.append(T)  # Append updated temperature to the list
    Number += 1

# Plotting the graph
print(T_values)
plt.plot(T_values)
plt.title("Temperature Variation Over Time")
plt.xlabel("Iteration")
plt.ylabel("Temperature (T)")
plt.grid(True)
plt.show()

File: Temperature simulator/Temperature2.js
const SENSITIVITY = 0.5;
const MAX_TEMP = 8;
const MIN_TEMP = 4;
let Number = 0; 
let T = Math.floor(Math.random() * (MAX_TEMP - MIN_TEMP + 1)) + MIN_TEMP;  // Initial random temperature

let T_values = [T];  // Array to store temperature values

while (Number < 100) {
    let newT = Math.floor(Math.random() * (MAX_TEMP - MIN_TEMP + 1)) + MIN_TEMP;

    while (T >= newT - SENSITIVITY) {
        T -= SENSITIVITY*0.5;
    }
    while (T <= newT - SENSITIVITY) {
        T += SENSITIVITY*0.5;
    }

    T_values.push(T);  // Append updated temperature to the array
    Number += 1;
}

console.log(T_values);  // Print the temperature values to the console
